Fix Placeholder in Cast on Details and Actor Search Screens

skaik-mo opened the pull request:
- Modify `PersonCacheDto.toEntity`, `Person.toDto`, `PersonDto.toEntity`, `CastDto.toEntity`, `CrewDto.toEntity`, and `PersonCreditDto.toEntity` to handle potential null image URLs by checking if the URL already contains `BASE_IMAGE_URL` before prepending it.
- In `SafeIslamicImage`, check if `imageUrl` is blank or "null" and set `isPlaceholder` to true accordingly to prevent loading issues.
- Update `StarCastSection` and `CastItem` to display an icon placeholder when the image URL is null or empty.
- Delete the `ResultsActors` composable as it is no longer used.

<img width="510" height="1005" alt="image" src="https://github.com/user-attachments/assets/44694a49-8343-464e-9530-dabd3fb897bb" />

<img width="510" height="1005" alt="image" src="https://github.com/user-attachments/assets/9d3de6e4-87b7-47f1-bd84-ba92a1e93381" />

skaik-mo committed:
fix: handle null image URLs

- Modify `PersonCacheDto.toEntity`, `Person.toDto`, `PersonDto.toEntity`, `CastDto.toEntity`, `CrewDto.toEntity`, and `PersonCreditDto.toEntity` to handle potential null image URLs by checking if the URL already contains `BASE_IMAGE_URL` before prepending it.
- In `SafeIslamicImage`, check if `imageUrl` is blank or "null" and set `isPlaceholder` to true accordingly to prevent loading issues.
- Update `StarCastSection` and `CastItem` to display an icon placeholder when the image URL is null or empty.
- Delete the `ResultsActors` composable as it is no longer used.

Mohamed-Ashraf-2210 commented on presentation/details/src/main/java/com/giraffe/details/components/StarCastSection.kt:
```diff
@@ -115,7 +118,17 @@ fun CastCard(
+            ) {
+                Icon(
+                    painter = painterResource(Theme.icons.dueTone.image),
+                    contentDescription = null,
```
add content description

skaik-mo committed:
fix: update content description for actor image in StarCastSection

skaik-mo committed:
Merge branch 'develop' into fix/placeholder-in-cast-on-details-and-actor-search-screens

skaik-mo committed:
Refactor: Rename `toDto` to `toCacheDto` in Person mappers and repository

This commit refactors the naming of the `toDto` extension function in the `Person` and `PersonCacheDto` mappers to `toCacheDto`. This change clarifies the function's purpose, indicating that it specifically converts to a `PersonCacheDto` for caching purposes.

The following files were updated to reflect this change:
- `repository/media/src/main/java/com/giraffe/media/person/mapper/Mapper.kt`
- `repository/media/src/main/java/com/giraffe/media/utils/PersonContentType.kt`
- `repository/media/src/main/java/com/giraffe/media/person/PersonRepositoryImpl.kt`

dev-bilal-azzam commented on imageviewer/src/main/java/com/giraffe/imageviewer/component/SafeIslamicImage.kt:
```diff
@@ -66,6 +66,10 @@ fun SafeIslamicImage(
     LaunchedEffect(imageUrl) {
         isPlaceholder = true
+        if (imageUrl.isBlank() || imageUrl == "null") {
+            isPlaceholder = true
```
no need

mohamedebrahem13 commented on presentation/details/src/main/java/com/giraffe/details/components/StarCastSection.kt:
```diff
@@ -115,7 +118,17 @@ fun CastCard(
+                    tint = Theme.color.brand.secondary,
+                    modifier = it
+                        .size(64.dp)
+                        .size(28.dp)
```
Why you are using size twice?

mohamedebrahem13 commented on presentation/explore/src/main/java/com/giraffe/explore/components/CastItem.kt:
```diff
@@ -42,7 +46,23 @@ fun CastItem(
+                        shape = RoundedCornerShape(Theme.radius.lg)
+                    )
+                    .size(28.dp)
+                    .wrapContentSize(),
```
I think You don't need Wrap content size because you give size before it

mohamedebrahem13 reviewed (changes requested):
You need to remove some extra modifiers

skaik-mo committed:
fix: adjust actor image size in StarCastSection for better layout

skaik-mo committed:
fix: remove redundant placeholder assignment in SafeIslamicImage

skaik-mo committed:
fix: remove redundant size assignment in CastItem and StarCastSection
